solutions/USACO feb21-clockwise-fence.mdx

```diff
@@ -0,0 +1,46 @@
+---
+id: usaco-1109
+source: USACO Bronze 2021 February
+title: Clockwise Fence
+author: Melody Yu
+---
+
+[Official Analysis (C++)](http://usaco.org/current/data/sol_prob3_bronze_feb21.html)
+
+## Solution (Vector Cross Product)
+
+The vector cross product can be used to determine if 2 vectors are rotating clockwise or anticlockwise. For this question, we treat each step that Farmer John walks as a vector and take the cross product of the current step and the previous step. If we sum all the cross product values, we can determine if the overall direction is CW or CCW based on the sign of the sum.
```

Comment:
(oops my math is kind of bad) we're only able to sum all the cross product values and check the sign because the vector lengths are the same, right? I think

> If we sum all the cross product values, we can determine if the overall direction is CW or CCW based on the sign of the sum.

could use with a bit of elaboration as to why this works (not why cross product returns positive or negative, but why we can sum together the cross product values and still do the positive/negative check). (Or maybe I'm just really bad at math oops, does anybody else have an opinion?)

Reply:
i think taking the cross products and summing them is essentially the same thing as shoelace?

Reply:
Basically cross product is twice the area of two vectors (positive/negative). By adding them together we get 2x the final area of the polygon formed by these vectors. So, based on the positive/negative value of the area, we can know if it's counterclockwise or clockwise (vector up or vector down). The process is how the shoelace formula is proved.

Should I add that then?

Reply:
sure, *twice the area of the triangle between the two vectors